File: admin-web/src/constants/levels.js
/** Audience / year levels — keep in sync with backend/src/constants/levels.js */

export const ELECTION_LEVELS = [
  { id: 'grade_7', label: 'Grade 7' },
  { id: 'grade_8', label: 'Grade 8' },
  { id: 'grade_9', label: 'Grade 9' },
  { id: 'grade_10', label: 'Grade 10' },
  { id: 'grade_11', label: 'Grade 11' },
  { id: 'grade_12', label: 'Grade 12' },
  { id: 'college_1', label: 'College — 1st Year' },
  { id: 'college_2', label: 'College — 2nd Year' },
  { id: 'college_3', label: 'College — 3rd Year' },
  { id: 'college_4', label: 'College — 4th Year' },
  { id: 'college_5', label: 'College — 5th Year' },
]

export function labelForLevel(id) {
  if (id === 'junior_high') return 'Junior High'
  if (id === 'college') return 'College'
  return ELECTION_LEVELS.find((l) => l.id === id)?.label || id
}

File: student-mobile/lib/utils/levels.dart
/// Keep labels in sync with admin-web / backend level ids.
const Map<String, String> kLevelLabels = {
  'grade_7': 'Grade 7',
  'grade_8': 'Grade 8',
  'grade_9': 'Grade 9',
  'grade_10': 'Grade 10',
  'grade_11': 'Grade 11',
  'grade_12': 'Grade 12',
  'college_1': 'College — 1st Year',
  'college_2': 'College — 2nd Year',
  'college_3': 'College — 3rd Year',
  'college_4': 'College — 4th Year',
  'college_5': 'College — 5th Year',
  'junior_high': 'Junior High',
  'college': 'College',
};

String labelForLevel(String? id) {
  if (id == null || id.isEmpty) return 'Not set';
  return kLevelLabels[id] ?? id;
}
